City-of-Helsinki/employment-services-ui
# THF-637: Added new event_index to event components

### Changes
Changed event and tag fetches to use new event_index. Added language filter


- Testing with branch https://github.com/City-of-Helsinki/drupal-employment-services/pull/177


### Testing
- http://localhost:3000/ajankohtaista/tapahtumat

#### Language select

- Test first Select language.
- When selecting a Drupal language (fi, en, sv, ru, so, ua), the tags should change to the selected language, and you should receive the same data as you would on the local page. For example, if you are on Finnish pages and you choose English in the filter, you should receive the same data as you would on an English page.
- Additionally, the lang tag on URL  should also change to the selected language. 


#### Select Topic
- Keep a language selected and add topics.
- The "Select Topic" functionality should work as before.
- The URL tags should always be in English, even if your tags are displayed in Finnish on the UI.

#### Test Memory

- Select some filters. 
- Then select an event and open the event on the same page.
- Go back with the browser back button or breadcrumbs.
- You should end up in the same place with the same filters you left.
-
- Select again some filter. 
- Choose some filters again. Choose a Drupal language that is different from your local language.
- For example, if you are on a Finnish page, choose English.
- Then, select an event and open the event on the same page.
- Go back using the browser back button.
- You should end up on the same local page as you left.
- Go back to the event and use breadcrumbs to return to the Event page.
- You should end up on the local page of the language you had before on the language filter.
- The same filters should also be in place.

#### Test URL
- Select a language and some topics. Copy the URL and open a new tab. Paste the copied URL.
- You should now have the same filters as you had when you left the page.
